crates/pallets/starknet/src/lib.rs

```diff
@@ -1166,6 +1107,77 @@ impl<T: Config> Pallet<T> {
         Ok(results)
     }
 
+    pub fn simulate_transactions(
+        transactions: Vec<UserTransaction>,
+        simulation_flags: Vec<SimulationFlag>,
+    ) -> Result<Vec<SimulatedTransaction>, DispatchError> {
+        let chain_id = Self::chain_id();
+
+        let disable_fee_charge = simulation_flags.contains(&SimulationFlag::SkipFeeCharge);
+        let disable_validation = simulation_flags.contains(&SimulationFlag::SkipValidate);
+
+        let execution_results = execute_txs_and_rollback::<T>(
+            &transactions,
+            &Self::get_block_context(),
+            chain_id,
+            false,
+            disable_validation,
+            disable_fee_charge,
+        );
+
+        let get_function_invocation =
+            |call_info: Option<&CallInfo>| -> Result<Option<FunctionInvocation>, DispatchError> {
+                match call_info {
+                    Some(call_info) => Ok(Some(convert_call_info_to_function_invocation::<T>(call_info)?)),
+                    None => Ok(None),
+                }
+            };
```

Comment:
If you don't need to capturate var in the outer scope, I would give my favor to `fn` rather than closure here

Reply:
fixed